src/ui/services/ipc/ipcUiService.ts

```diff
@@ -62,49 +60,27 @@ export class IpcUiService {
     this.mainSendMessage(IpcMainChannel.MAIN_IPC_REQUEST_CHANNEL_REFRESH, null);
   }
 
-  private async onSetWorkerChannel(workerChannel: MessagePort): Promise<void> {
+  private async setNewWorkerChannel(workerChannel: MessagePort): Promise<void> {
     // Save the new channel.
     this.mainLog('Got new worker channel.');
     this.workerChannel = workerChannel;
-
-    // Listen for messages
-    workerChannel.onmessage = async (workerChannelEvent) => {
-      const workerMessage = workerChannelEvent.data as IpcUiMessage;
-      if (this.onUiMessage !== null) {
-        const resp = await this.onUiMessage(workerMessage);
-        this.mainLog(`[TODO] Unsent response: ${JSON.stringify(resp)}`);
-        return;
-      }
-      this.mainLog(`Got unhandled worker message: ${workerMessage}`);
-    };
-
-    // Start the port
-    workerChannel.start();
-
-    // Say hi.
-    const resp: IpcWorkerResponseOf<IpcWorkerMessageType.UTILS_PING> =
-      await this.workerInvoke({
-        type: IpcWorkerMessageType.UTILS_PING,
-        data: {
-          message: 'Hello from UI',
-        },
-      });
-    this.mainLog(`Said hi to worker, he replied ${resp.message}.`);
+    this.onNewWorkerChannel.emit();
   }
 
   private workerSendMessage(
     message: IpcWorkerMessage,
     transferables?: Transferable[]
   ): void {
     if (this.workerChannel === null) {
-      throw new Error(
-        'Tried to send message before the workerChannel was initiated'
-      );
+      this.onNewWorkerChannel.once(() => {
```

Comment:
This function never throws now.

Instead of raising, just wait until there is a channel before sending the message.